### DISCUSSION TITLE
[DataTable] Reorder feature and rowreordericon slot problem.

### DISCUSSION THREAD

**Discussion by tuffkidd** [primefaces/community#42] [PrimeVue] [answered]
The Reorder feature of DataTable sometimes doesn't work correctly.

1. When I click the reorder icon, the draggable attribute of the corresponding <tr> changes from false to true.
2. However, after moving it, occasionally the false does not change back to true, making it impossible to reorder.
3. After randomly clicking other areas and then clicking it again, it changes to true.

![PrimeVue-DataTable-Reorder](https://github.com/primefaces/community/assets/2167254/982b31e2-aef3-4f3b-aafc-0ffb18f6697d)

And when using the rowreordericon slot, the icon appears but it doesn't work. See the code below.
```html
<Column :row-reorder="true" header-style="width: 3em">
  <template #rowreordericon>
    <i :class="['p-datatable-reorderablerow-handle', 'ICONCLASS']" style="font-size: 1.25rem"></i>
  </template>
</Column>
```

**Comment by atkntepe** [comment]
Hey there, thanks for letting us know. I opened up an issue for this. You can follow it from here https://github.com/primefaces/primevue/issues/4459

**Comment by tugcekucukoglu** [comment, marked as answer]
There is a bug on the mouse-down event and it's fixed for the next version. Also `rowreordericon` slot provides the necessary class, you don't have to type manually, just be added like below.

```
<Column rowReorder :reorderableColumn="false">
    <template #rowreordericon="slotProps">
            <i :class="['ICONCLASS', slotProps.class]" data-pc-section="rowreordericon" />
    </template>
</Column>
```

PrimeVue changes `draggable` attribute when the mouse-down is fired not the click event. Because there is a drag-drop process, it did not seem appropriate to replace it with a click event.

Moreover I want to this from MDN documentation:
```
Note: This differs from the [click](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) event in that click is fired after a full click action occurs; that is, the mouse button is pressed and released while the pointer remains inside the same element. mousedown is fired the moment the button is initially pressed.
```

**Reply by tuffkidd** [reply]
Thank you for fixing it quickly. I really love PrimeVue.
